src/http/request.cr

```diff
@@ -98,6 +98,16 @@ class HTTP::Request
     value
   end
 
+  # Return request host from headers
+  def host
+    @headers["Host"]?.try &.to_s.sub(/:\d+\z/, "")
```

Comment:
Yes, using Regex is almost always slower than not using Regex. Specially if this is just a char index lookup. Matching a regex involves much more logic than just a char index, and also allocated memory for the match results (and I don't know if PCRE allocates memory for that too).

Try this:

```cr
require "benchmark"

def regex(string)
  string.sub(/:\d+\z/, "")
end

def index(string)
  index = string.index(':')
  index ? string[0...index] : string
end

# This is so LLVM can't optimize it that easily
string = String.build { |str| str << "www.google.com:80" }

pp regex(string)
pp index(string)

Benchmark.ips do |x|
  x.report("regex") { regex(string) }
  x.report("index") { index(string) }
end
```

I get:

```
regex(string) # => "www.google.com"
index(string) # => "www.google.com"
regex   1.41M (±14.83%)  4.38× slower
index   6.18M (±15.03%)       fastest
```